### ISSUE TITLE
Syntax error when using special characters in formData key

### ISSUE THREAD

**Issue by lmichelin** [anymaniax/orval#584]
## What are the steps to reproduce this issue?

Having an OpenAPI spec containing a POST route using multipart/form-data and whose body contains a key with special characters:

<details>
<summary>openapi-spec.json</summary>

```
{
  "openapi": "3.0.3",
  "info": {
    "title": "",
    "description": "",
    "version": "0.0.0"
  },
  "servers": [
    {
      "url": "/",
      "description": ""
    }
  ],
  "paths": {
    "/api/test_object": {
      "post": {
        "operationId": "postTestObjectCollection",
        "tags": ["TestObject"],
        "responses": {
          "201": {
            "description": "TestObject resource created",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/TestObject-read.collection.test_object"
                }
              }
            },
            "links": {
              "GetTestObjectItem": {
                "operationId": "getTestObjectItem",
                "parameters": {
                  "id": "$response.body#/id"
                },
                "description": "The `id` value returned in the response can be used as the `id` parameter in `GET /api/test_object/{id}`."
              }
            }
          },
          "400": {
            "description": "Invalid input"
          },
          "422": {
            "description": "Unprocessable entity"
          }
        },
        "summary": "Creates a TestObject resource.",
        "description": "Creates a TestObject resource.",
        "parameters": [],
        "requestBody": {
          "description": "The new TestObject resource",
          "content": {
            "multipart/form-data": {
              "schema": {
                "$ref": "#/components/schemas/TestObject.CreateTestObjectInput"
              }
            }
          },
          "required": true
        },
        "deprecated": false
      },
      "parameters": []
    }
  },
  "components": {
    "schemas": {
      "TestObject-read.collection.test_object": {
        "type": "object",
        "required": ["relatedEntity"],
        "properties": {
          "relatedEntity": {
            "type": "string",
            "format": "iri-reference"
          },
          "uploadedFiles": {
            "type": "array",
            "items": {
              "type": "string"
            }
          }
        }
      },
      "TestObject.CreateTestObjectInput": {
        "type": "object",
        "required": ["relatedEntityId"],
        "properties": {
          "relatedEntityId": {
            "format": "uuid",
            "externalDocs": {
              "url": "http://schema.org/identifier"
            },
            "type": "string"
          },
          "files[]": {
            "type": "array",
            "items": {
              "type": "string",
              "format": "binary"
            }
          }
        }
      }
    },
    "responses": {},
    "parameters": {},
    "examples": {},
    "requestBodies": {},
    "headers": {},
    "securitySchemes": {
      "apiKey": {
        "type": "apiKey",
        "description": "Value for the Authorization header parameter.",
        "name": "Authorization",
        "in": "header"
      }
    }
  },
  "security": [
    {
      "apiKey": []
    }
  ],
  "tags": []
}
```

</details>

<details>
<summary>orval.config.js</summary>

```
module.exports = {
  api: {
    output: {
      target: "src/networking/api.ts",
      client: "react-query",
      prettier: true,
    },
    input: {
      target: "./openapi-spec.json",
    },
  },
}
```

</details>

## What happens?

Generated code contains a syntax error and can neither be executed not formatted with prettier:

![image](https://user-images.githubusercontent.com/33673240/190187064-2627599c-dc87-439f-a99d-bcb846d1c390.png)

## What were you expecting to happen?

`testObjectCreateTestObjectInput.files[]` should be replaced with `testObjectCreateTestObjectInput['files[]']`:

![image](https://user-images.githubusercontent.com/33673240/190188068-0f33127f-dafb-4786-9aab-9eb0b0c05bbf.png)

## What versions are you using?

**Operating System:** Ubuntu
**Package Version:** 6.9.6

**Comment by anymaniax** [comment]
Will be fixed in the next release

**Comment by anymaniax** [comment]
should be fixed with `6.10.2`

**Comment by lmichelin** [comment]
I can confirm that the issue is fixed in version `6.10.2`, thanks! 🎉
